src/Missings.jl

```diff
@@ -215,11 +215,17 @@ end
 function (f::SpreadMissings{F})(xs...; kwargs...) where {F}
     if any(x -> x isa AbstractVector{>:Missing}, xs)
         vecs = Base.filter(x -> x isa AbstractVector, xs)
-    nonmissingmask = fill(true, length(vecs[1]))
-    for v in vecs
-        nonmissingmask .&= .!ismissing(v)
-    end
-    nonmissinginds = findall(nonmissingmask)
+
+        findex = eachindex(first(vecs))
+        @assert all(x -> eachindex(x) == findex, vecs)
```

Comment:
Can you throw a more explicit error in case of failure? It would be nice to mention the indices which do not match.